packages/python/src/armonik/client/events.py

```diff
@@ -56,66 +57,84 @@ def get_events(
             event_types: The list of the types of event to catch.
             event_handlers: The list of handlers that process the events. Handlers are evaluated in he order they are provided.
                 An handler takes three positional arguments: the ID of the session, the type of event and the event as an object.
-                An handler returns a boolean, if True the process continues, otherwise the stream is closed and the service stops
-                listening to new events.
+                An handler returns a boolean, if False the process continues, otherwise the stream is closed and the service stops
+                listening to new events. As handlers are evaluated in order, when a handler interrupts execution (by returning True)
+                all the handlers following it will not be executed.
             task_filter: A filter on tasks.
             result_filter: A filter on results.
 
         """
-        request = EventSubscriptionRequest(session_id=session_id, returned_events=event_types)
-        if task_filter:
-            request.tasks_filters = cast(rawTaskFilters, task_filter.to_disjunction().to_message())
-        if result_filter:
-            request.results_filters = cast(
-                rawResultFilters, result_filter.to_disjunction().to_message()
-            )
+        request = EventSubscriptionRequest(
+            session_id=session_id,
+            returned_events=event_types,
+            tasks_filters=cast(rawTaskFilters, task_filter.to_disjunction().to_message())
+            if task_filter
+            else rawTaskFilters(),
+            results_filters=cast(rawResultFilters, result_filter.to_disjunction().to_message())
+            if result_filter
+            else rawResultFilters(),
+        )
 
         streaming_call = self._client.GetEvents(request)
         for message in streaming_call:
             event_type = message.WhichOneof("update")
-            if any(
-                [
-                    event_handler(
-                        session_id,
-                        EventTypes.from_string(event_type),
-                        self._events_obj_mapping[event_type].from_raw_event(
-                            getattr(message, event_type)
-                        ),
-                    )
-                    for event_handler in event_handlers
-                ]
-            ):
-                break
-
-    def wait_for_result_availability(self, result_id: str, session_id: str) -> None:
+            for event_handler in event_handlers:
+                if event_handler(
+                    session_id,
+                    EventTypes.from_string(event_type),
+                    self._events_obj_mapping[event_type].from_raw_event(
+                        getattr(message, event_type)
+                    ),
+                ):
+                    break
+
+    def wait_for_result_availability(self, result_ids: List[str], session_id: str) -> None:
```

Comment:
I would keep the old one too with a
```suggestion
    def wait_for_result_availability(self, result_ids: Union[str, List[str]], session_id: str) -> None:
```
And wrap the singular one in a list
for Api compatibility